### ISSUE TITLE
Lipschitz continuous conditions

### ISSUE THREAD

**Issue by EmreTaha** [master/nima#3]
As far as I know in order to be able to use wasserstein distance in dual form, which saves us from infimum, the function should be liptschitz-K continuous, so that the gradients don`t overshoot. Generally it is solved in deep learning by using, weight clipping, gradient penalty or spectral normalization. But in your case, is there a specific reason Why you didn`t take this into account?

Also I used your NIMA in another dataset with Resnet-50-v2 + softmax with your training schedule, I observed that sometimes loss becomes nan or inf. which I stabilized by adding a small number to loss (something like e-10), but it destabilized the network in later epochs. Have you ever encountered problems like this? Please let me know.

Thanks.
